// Copyright 2019 The MediaPipe Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";

package mediapipe;

import "mediapipe/framework/calculator.proto";
import "mediapipe/util/color.proto";

message RectToRenderDataCalculatorOptions {
  extend CalculatorOptions {
    optional RectToRenderDataCalculatorOptions ext = 262270380;
  }

  // Whether the rendered rectangle should be filled.
  optional bool filled = 1;

  // Line color or filled color of the rectangle.
  optional Color color = 2;

  // Thickness of the line (applicable when the rectangle is not filled).
  optional double thickness = 3 [default = 1.0];
}
